e2e/src/test/scala/scalapb/zio_grpc/TestServiceImpl.scala

```diff
@@ -45,32 +45,38 @@ package object server {
       ): ZStream[Any, Status, Response] =
         ZStream
           .bracketExit(requestReceived.succeed(())) { (_, ex) =>
-            ex.foldM(
+            ex.foldZIO(
               failed =>
-                if (failed.interrupted)
+                if (failed.isInterrupted || failed.isInterruptedOnly)
                   exit.succeed(Exit.fail(Status.CANCELLED))
                 else exit.succeed(Exit.fail(Status.UNKNOWN)),
               _ => exit.succeed(Exit.succeed(Response()))
             )
           }
-          .flatMap { _ =>
+          .zipRight {
```

Comment:
This breaks streaming responses for some reason. These two lines should be equiv since `_` is never actually consumed.